web/auth/flagship.go

```diff
@@ -164,3 +172,87 @@ func confirmFlagship(c echo.Context) error {
 	}
 	return c.NoContent(http.StatusNoContent)
 }
+
+type loginFlagshipParameters struct {
+	ClientID          string `json:"client_id"`
+	ClientSecret      string `json:"client_secret"`
+	Passphrase        string `json:"passphrase"`
+	TwoFactorPasscode string `json:"two_factor_passcode"`
+	TwoFactorToken    string `json:"two_factor_token"`
+}
+
+func loginFlagship(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+
+	var args loginFlagshipParameters
+	if err := c.Bind(&args); err != nil {
+		return jsonapi.Errorf(http.StatusBadRequest, "%s", err)
+	}
+
+	if lifecycle.CheckPassphrase(inst, []byte(args.Passphrase)) != nil {
+		err := limits.CheckRateLimit(inst, limits.AuthType)
+		if limits.IsLimitReachedOrExceeded(err) {
+			if err = LoginRateExceeded(inst); err != nil {
+				inst.Logger().WithNamespace("auth").Warn(err.Error())
+			}
+		}
+		return c.JSON(http.StatusUnauthorized, echo.Map{
+			"error": inst.Translate(CredentialsErrorKey),
+		})
+	}
+
+	if inst.HasAuthMode(instance.TwoFactorMail) {
+		if len(args.TwoFactorPasscode) == 0 || len(args.TwoFactorToken) == 0 {
+			twoFactorToken, err := lifecycle.SendTwoFactorPasscode(inst)
+			if err != nil {
+				return err
+			}
+			return c.JSON(http.StatusAccepted, echo.Map{
```

Comment:
Why do we return `http.StatusAccepted` instead of `http.StatusForbidden` in `CreateSessionCode`? Should we homogenise those scenario?

Reply:
Yep, it has been fixed to `http.StatusForbidden`.